### ISSUE TITLE
Feature request: Alert for exporting an array

### ISSUE THREAD

**Issue by andras-tim** [koalaman/shellcheck#901]
#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
array=()

function set_array()
{
    export array=(1 2)
}

set_array
echo "${array[*]}"
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
Bash doesn't support exporting an array

#### Extra info
StackOverflow question about this:
http://stackoverflow.com/questions/5564418/exporting-an-array-in-bash-script

**Comment by NorthboundTrain** [comment]
The current maintainer of Bash (Jan 2022) said in 2014 that he had no intention of every supporting the export of Bash arrays:

https://web.archive.org/web/20150530054828/https://www.openwall.com/lists/oss-security/2014/09/30/1

Seems like it would be a great thing for shellcheck to look for and prevent.

**Comment by meleu** [comment]
yes please! :pray: 

Add an alert warning for when trying to export an array!

**Comment by andrewmcdonald-oxb** [comment]
I was just bitten by this, as I didn't realise arrays wouldn't export. If Shellcheck can't detect this currently, it would be great to have.
